### PR TITLE
fix: #335 Fix modal and used it instead of confirm

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/component/modal/Modal.stories.tsx [hunk 1]
```diff
@@ -32,7 +32,8 @@ const Template: StoryFn<typeof Modal> = (args) => {
       <Modal
         content={defaultContent}
         isOpen={isOpen}
-        handleClose={() => setIsOpen(false)}
+        text="Okey"
```
Comment (tritonJS826) [inline mw-webapp/src/component/modal/Modal.stories.tsx:35]: okText

File: mw-webapp/src/logic/wayPage/MentorsSection.tsx [hunk 2]
```diff
@@ -86,18 +90,18 @@ export const MentorsSection = (props: MentorsSectionProps) => {
               >
                 <TrashIcon
                   className={styles.icon}
-                  onClick={() => {
+                  onClick={() => setModalElementUuid(mentor.uuid)}
+                />
+                {isModalElementUuidExist && modalElementUuid === mentor.uuid &&
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/MentorsSection.tsx:95]: Could we simlify this line

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 3]
```diff
@@ -158,6 +158,8 @@ export const WayPage = (props: WayPageProps) => {
   const [isStatisticsVisible, setIsStatisticsVisible] = useState<boolean>(isCurrentStatisticsVisible);
   const {user, setUser} = useGlobalContext();
   const [way, setWay] = useState<Way>();
+  const [modalElementUuid, setModalElementUuid] = useState<string>();
+  const isModalElementUuidExist = !!modalElementUuid;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:162]: do we need this variable

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 4]
```diff
@@ -222,6 +224,14 @@ export const WayPage = (props: WayPageProps) => {
     setIsStatisticsVisible(!isStatisticsVisible);
   };
 
+  /**
+   * Delete way
+   */
+  const deleteWay = async () => {
+    await WayDAL.deleteWay(way);
+    user && navigate(pages.user.getPath({uuid: user.uuid}));
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:232]: maybe

**Review by tritonJS826 — APPROVED** [review]

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/logic/wayPage/GoalMetricItem.tsx [hunk 5]
```diff
@@ -161,16 +161,14 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
-              /**
-               * CallBack remove singular goal metric
-               */
-              onOk: () => removeSingularGoalMetric(props.singleGoalMetric.metricUuid),
-            })}
+          <Confirm
+            trigger={
+              <TrashIcon className={styles.icon} />}
+            content={<p>
+              {`Are you sure that you want to delete singleGoalMetric "${props.singleGoalMetric.description}"?`}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/GoalMetricItem.tsx:168]:
```suggestion
              {`Are you sure you want to delete goal metric "${props.singleGoalMetric.description}"?`}
```

File: mw-webapp/src/logic/wayPage/MentorsSection.tsx [hunk 6]
```diff
@@ -84,19 +85,14 @@ export const MentorsSection = (props: MentorsSectionProps) => {
...
-
-                    // TODO: use modal instead of confirm task #305
-                    const isConfirmed = confirm(`Are you sure you want remove "${mentor.name}" from mentors?`);
-                    isConfirmed && onOk();
-                  }}
+                <Confirm
+                  trigger={
+                    <TrashIcon className={styles.icon} />}
+                  content={<p>
+                    {`Are you sure that you want to to remove "${mentor.name}" from mentors"?`}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/MentorsSection.tsx:92]:
```suggestion
                    {`Are you sure you want to remove "${mentor.name}" from mentors"?`}
```

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 7]
```diff
@@ -268,22 +276,18 @@ export const WayPage = (props: WayPageProps) => {
...
-            }
+          <Confirm
+            trigger={
+              <Button
+                value="Delete way"
+                buttonType={ButtonType.TERTIARY}
+                onClick={() => {}}
+              />}
+            content={<p>
+              {`Are you sure that you want to delete way "${way.name}"?`}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:287]:
```suggestion
              {`Are you sure you want to delete the way "${way.name}"?`}
```

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 8]
```diff
@@ -132,6 +108,13 @@ const updateDayReportState = (
   setReports(updatedDayReports);
 };
 
+/**
+ * Case when need render confirm
+ */
+export const isRenderConfirm = (confirmUuid: string | undefined, uuid: string) => {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:114]: it is not boolean -> getIsRenderConfirm

Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:114]: not clear what the difference betwee uuid an dconfirm uuid and why do we need them both

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 9]
```diff
@@ -751,17 +723,13 @@ export const Columns = (props: ColumnsProps) => {
...
-                            /**
-                             * CallBack triggered on press ok
-                             */
-                            onOk: () => deleteComment(comment.uuid),
-                          })
-                          }
+                        <Confirm
+                          trigger={<TrashIcon className={styles.icon} />}
+                          content={<p>
+                            {`Are you sure that you want to delete comment "${comment.description}"?`}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:729]:
```suggestion
                            {`Are you sure you want to delete the comment "${comment.description}"?`}
```

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx (hunk 8 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:114]: Deleted

Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:114]: Deleted deprecated code

**Pull request merged**